lib/travis/build/script/shared/directory_cache/base.rb

```diff
@@ -109,8 +109,10 @@ def install
             sh.export 'CASHER_DIR', '$HOME/.casher'
 
             sh.mkdir '$CASHER_DIR/bin', echo: false, recursive: true
-            sh.cmd "curl #{casher_url} #{debug_flags} -L -o #{BIN_PATH} -s --fail", retry: true, echo: 'Installing caching utilities'
-            sh.raw "[ $? -ne 0 ] && echo 'Failed to fetch casher from GitHub, disabling cache.' && echo > #{BIN_PATH}"
+            update_static_file('casher', BIN_PATH, casher_url, true)
+            sh.if "$? -ne 0" do
```

Comment:
Here, `$?` should be non-zero exactly when _all_ of the `curl` commands (1 or 2, depending on `casher_branch == 'production'`) exit with non-zero status.